qualtran/bloqs/mod_arithmetic/mod_addition.py

```diff
@@ -339,3 +342,109 @@ def wire_symbol(
         if reg.name == 'y':
             return TextBox(f'y += x*{self.k}')
         raise ValueError(f"Unknown register {reg}")
+
+
+@frozen
+class CModAdd(Bloq):
+    r"""Controlled Modular Addition.
+
+    Implements $\ket{c}\ket{x}\ket{y} \xrightarrow[]{} \ket{c}\ket{x}\ket{(cx+y)\%p}$
+    using $5n+1$ Toffoli gates.
+
+    Note: The true toffoli count is $5n$ however we have an extra toffoli
+    due to how our OutOfPlaceAdder works.
```

Comment:
I'm not wild about the phrase "true toffoli count". Can you re-word this.

Note: the reference reports 5n toffolis. Our construction has an extra toffoli gate due to the current implementaiton of `OutOfPlaceAdder`

or something similar

Reply:
updated